octree/include/pcl/octree/octree_pointcloud_adjacency_container.h

```diff
@@ -117,10 +117,10 @@ class OctreePointCloudAdjacencyContainer : public OctreeContainerBase {
   /** \brief  virtual method to get size of container
    * \return number of points added to leaf node container.
    */
-  std::size_t
+  index_t
```

Comment:
uindex_t? and remove unnecssary cast as num_points_ is uindex_t?

Reply:
getPointCounter and getSize returns the same variable value? Deprecate getPointCounter? (maybe another pr)

Reply:
OctreePointCloudDensity has a getPointCounter but not getSize. I agree in principle (someone should do it)